api/runner/runner.go

```diff
@@ -29,6 +29,7 @@ type Runner struct {
 	flog         FuncLogger
 	availableMem int64
 	usedMem      int64
+	Tasks        chan task.Request
```

Comment:
> Like you you've noticed the channel is still visible and can be easily replaced externally.

Shouldn't code be closed for modification and open for extension? Or putting differently, if you are exposing an internal part like this, you are also saying for future customizers of IronFunction that a) Tasks can be replaced at any time (which his not true); b) Tasks do not need any protection when being changed (channel itself is safe, but if you can just replace the channel with something else, then you could have a race condition here).

> If you take a look on tasks no change has been done to the test it self, we are currently testing the same way as before, but the channel creation is no longer exposed.

That's not true - or I might have read your changes incorrectly. With the external channel, there was mocks that pretended that tasks had been consumed. In this PR, you are just running these tasks in the Docker again, no?

> So why not embedding that to it?

I clearly failed to convey my concern, and I apologize. You are just stating that your change is better (embedding), but I did not see anywhere why before was worse. You essentially took a channel that was meant to link together two different components, and put in one of them - in practice defeating the meaning of it being external: show the connection of two independent executing processes.

> also reduce the complexity for the user when he wants to creates his own custom IronFunctions server.

Could you please elaborate on this? It seems something important. Perhaps some example?